cobalt/common/libc/tz/parse_posix_tz_test.cc

```diff
@@ -0,0 +1,243 @@
+// Copyright 2025 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "cobalt/common/libc/tz/parse_posix_tz.h"
+
+#include <ostream>
+#include <string>
+#include <vector>
+
+#include "gtest/gtest.h"
+
+namespace cobalt {
+namespace common {
+namespace libc {
+namespace tz {
+
+// Number of seconds in an hour.
+constexpr int kHour = 3600;
+// Number of seconds in a minute.
+constexpr int kMinute = 60;
+
+// --- Equality Operators for Comparison ---
+bool operator==(const DateRule& lhs, const DateRule& rhs) {
+  return lhs.format == rhs.format && lhs.month == rhs.month &&
+         lhs.week == rhs.week && lhs.day == rhs.day;
+}
+
+bool operator==(const TransitionRule& lhs, const TransitionRule& rhs) {
+  return lhs.date_rule == rhs.date_rule && lhs.time == rhs.time;
+}
+
+bool operator==(const TimezoneData& lhs, const TimezoneData& rhs) {
+  return lhs.std_name == rhs.std_name && lhs.std_offset == rhs.std_offset &&
+         lhs.dst_name == rhs.dst_name && lhs.dst_offset == rhs.dst_offset &&
+         lhs.start_rule == rhs.start_rule && lhs.end_rule == rhs.end_rule;
+}
+
+// Structure to hold all parameters for a single test case.
+struct TzParseTestParam {
+  std::string test_name;
+  std::string tz_string;
+  std::optional<TimezoneData> expected_result;
+
+  // Allows gtest to print readable output.
+  friend void PrintTo(const TzParseTestParam& param, std::ostream* os);
+};
+
+// `PrintTo` function for `TzParseTestParam` so that gtest will print readable
+// output when a test fails. This version prints on a single line.
+void PrintTo(const TzParseTestParam& param, std::ostream* os) {
+  *os << "{ test_name: \"" << param.test_name << "\", "
+      << "tz_string: \"" << param.tz_string << "\", "
+      << "expected_result: ";
+  if (param.expected_result.has_value()) {
+    const auto& data = *param.expected_result;
+    *os << "{ std_name: \"" << data.std_name << "\", "
+        << "std_offset: " << data.std_offset;
+    if (data.dst_name && !data.dst_name->empty()) {
+      auto print_rule = [&](const TransitionRule& rule) {
```

Comment:
The PrintTo function become more readable if print_rule is made its own function instead of being a llambda